Update sequence functions docs (#17713)
pingcap/docs: functions-and-operators/sequence-functions.md

Sequence functions in TiDB are used to return or set values of sequence objects created using the [`CREATE SEQUENCE`](/sql-statements/sql-statement-create-sequence.md) statement.

| Function name | Description |
| :-------------- | :------------------------------------- |
| [`NEXTVAL()`](#nextval) | Returns the next value of a sequence. |
| [`NEXT VALUE FOR`](#next-value-for) | Returns the next value of a sequence (alias for `NEXTVAL()`). |
| [`SETVAL()`](#setval) | Sets the current value of a sequence. |

> | [`LASTVAL()`](#lastval) | Returns the last used value of a sequence in the current session. |

Reviewer: ```suggestion
| [`LASTVAL()`](#lastval) | Returns the last value generated by a sequence in the current session. |
```

|           2 |
+-------------+
1 row in set (0.00 sec)
```

Get the next value from `s1` using `NEXT VALUE FOR`:

```sql
SELECT NEXT VALUE FOR s1;
```

The output is as follows:

```
+-------------------+
| NEXT VALUE FOR s1 |
+-------------------+
|                 3 |
+-------------------+
1 row in set (0.00 sec)
```

## `SETVAL()`

The `SETVAL(n)` function sets the current value of a sequence.

Example:

Get the next value from `s1`:

```sql
SELECT NEXTVAL(s1);
```

The output is as follows:

```
+-------------+
| NEXTVAL(s1) |
+-------------+
|           4 |
+-------------+
1 row in set (0.00 sec)
```

Set the current value of `s1` to `10`:

```sql
SELECT SETVAL(s1, 10);
```

The output is as follows:

```
+----------------+
| SETVAL(s1, 10) |
+----------------+
|             10 |
+----------------+
1 row in set (0.00 sec)
```

Verify the next value after setting it to `10`:

```sql
SELECT NEXTVAL(s1);
```

The output is as follows:

```
+-------------+
| NEXTVAL(s1) |
+-------------+
|          11 |
+-------------+
1 row in set (0.00 sec)
```

## `LASTVAL()`

The `LASTVAL()` function returns the last value generated by the sequence **in the current session**.

Example:

> Get the next value from `s1`:

Reviewer 2: ```suggestion
Get the last value generated by `s1` in the current session:
```

| Function name | Feature description |
| :-------------- | :------------------------------------- |
| [`NEXTVAL()`](#nextval) | Returns the next value of a sequence |
| [`NEXT VALUE FOR`](#next-value-for) | Returns the next value of a sequence |
| [`SETVAL()`](#setval) | Sets the current value of a sequence |
| [`LASTVAL()`](#lastval) | Returns the last used value of a sequence |

## `NEXTVAL()`

The `NEXTVAL()` functions returns the next value from the sequence

Example:

We create a sequence called `s1`.

```sql
CREATE SEQUENCE s1;
```

```
Query OK, 0 rows affected (0.01 sec)
```

Then we request the next value from `s1`.

```sql
SELECT NEXTVAL(s1);
```

```
+-------------+
| NEXTVAL(s1) |
+-------------+
|           1 |
+-------------+
1 row in set (0.00 sec)
```

## `NEXT VALUE FOR`

An alias for [`NEXTVAL()`](#nextval)

```sql
SELECT NEXTVAL(s1);
```

```
+-------------+
| NEXTVAL(s1) |
+-------------+

> |          13 |

Reviewer 3: Maybe use "2" instead of 13 here, so it follows the call flow (also updated the below?)

+-------------+
1 row in set (0.00 sec)
```

```sql
SELECT NEXT VALUE FOR s1;
```

```
+-------------------+
| NEXT VALUE FOR s1 |
+-------------------+
|                14 |
+-------------------+
1 row in set (0.00 sec)
```

## `SETVAL()`

The `SETVAL(n)` function can be use to set a sequence to a specific value.

```sql
CREATE SEQUENCE s1;
```

```
Query OK, 0 rows affected (0.01 sec)
```

Then we request the next value from `s1`.

```sql
SELECT NEXTVAL(s1);
```

```
+-------------+
| NEXTVAL(s1) |
+-------------+
|           1 |
+-------------+
1 row in set (0.00 sec)

> ```

Reviewer 3: I think we can re-use the same sequence for all statements here, but just keep them in order, so the values makes sense as if we where to issue them from top to bottom.

Now we set the value to `10`.

```sql
SELECT SETVAL(s1, 10);
```

```
+----------------+
| SETVAL(s1, 10) |
+----------------+
|             10 |
+----------------+
1 row in set (0.00 sec)
```

And now we can see that the next value (the value after `10`) is `11`.

```sql
SELECT NEXTVAL(s1);
```

```
+-------------+
| NEXTVAL(s1) |
+-------------+
|          11 |
+-------------+
1 row in set (0.00 sec)
```

## `LASTVAL()`

> The `LASTVAL()` function returns the last value that was generated by the sequence.

Reviewer 3: I think it should also be "generated by the sequence in **this session**", with 'this session' highlighted/bold text!

```sql
SELECT LASTVAL(s1);
```

```
+-------------+
| LASTVAL(s1) |
+-------------+

> |          12 |

Reviewer 3: Should not this be "11", same as last call to `NEXTVAL(s1)`?
